---
title: "European population X chr QC"
author: "Matthew Fischer"
date: "2025-01-22"
output: html_document
---

This code is how I performed X chromosome QC for the European population in MESA proteomics WGS data, can be adapted for other populations as needed.


##Keep plink tag to keep only european population:
#Using the post removing duplicated variants and updating sex from all populations RMD 
Initial input: ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/allpops/x-chromosome_updated-sex
Keep file: ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/final_qc_plinkfiles/europop/europop_final.fam

plink --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/allpops/x-chromosome_updated-sex --keep-fam ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/final_qc_plinkfiles/europop/europop_final.fam --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_initial

#Results:
4190194 variants loaded from .bim file.
2953 people (1447 males, 1506 females) loaded from .fam.
Total genotyping rate in remaining samples is 0.986339.
4190194 variants and 1270 people pass filters and QC.

##Check sex
plink --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_initial --check-sex --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/check_sex

#Results:
1270 people (621 males, 649 females) loaded from .fam.
--check-sex: 1263804 Xchr and 0 Ychr variant(s) scanned, no problems detected.

##Geno 0.02, maf 0.01
plink -bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_initial --geno 0.02 --maf 0.01 --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_geno_maf

#Results:
4190194 variants loaded from .bim file.
Total genotyping rate is 0.986339.
152107 variants removed due to missing genotype data (--geno).
3737430 variants removed due to minor allele threshold(s)
(--maf/--max-maf/--mac/--max-mac).
300657 variants and 1270 people pass filters and QC.

##Check sex post qc
plink --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_geno_maf --check-sex --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/check_sex_postfiltered

#Results
--check-sex: 300657 Xchr and 0 Ychr variant(s) scanned, 1 problem detected.

##Split-par
plink2 --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/europop_xchr_geno_maf --split-par b38 --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-split

#This command splits the X chromosome into:
    "PAR1"
    "X" (non-par)
    "PAR2"

#Results:
--split-par: 19019 chromosome codes changed.

##Perform HWE on PAR only regions:
plink2 --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-split --chr PAR1,PAR2 --hwe 1e-6 --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-hwe-par-only

#Results:
19019 out of 300657 variants loaded
--hwe: 2545 variants removed due to Hardy-Weinberg exact test (founders only).
16474 variants remaining after main filters.

##Write out non-PAR region plink files to merge with the hwe qcd par only data:
plink2 --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-split --chr X --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-only

#Results:
281638 out of 300657 variants loaded

##Combine post hwe Par only region with non-par region:
plink --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-hwe-par-only --bmerge ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-only --allow-extra-chr --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-post-hwe-merge

#Results (looks right):
298112 variants and 1270 people pass filters and QC.

##Recode the chr positions as X with --merge-x, out will be in the final_qc file 
```{r}
bimfile <- fread("~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-post-hwe-merge.bim", header = F, sep = "\t")

recode <- bimfile[V1 %in% c("PAR1", "PAR2"), V1 := "23"]

fwrite(recode, "~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-post-hwe-merge-recoded.bim", col.names = F, sep = "\t")
```

#Rename the new bim as the original and then create new plink files in final qc plink files folder 
plink --bfile ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/x-chromosome_qc/europop/split-par/europop-x-chr-post-hwe-merge --make-bed --out ~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/final_qc_plinkfiles/europop/europop_X-chr

#Results (looking into the bim with less -S and tail in command line, looks correct!:
Total genotyping rate is 0.999554.
298112 variants and 1270 people pass filters and QC.

##Final file path: 
~/2024-09-10_TOPMed_MESA_WGS_QC/QC_MESAproteomics_allchr/final_qc_plinkfiles/europop/europop_X-chr